### PR TITLE
feat(FEC-9479): auto focus on playlist preview popup and screen

### PR DESCRIPTION
**Pull request by RoyBregman**
### Description of the Changes

- refactored popup preview to another separate component so it is not rendered when not needed
- added aria-labels and roles to playlist components
- auto focused when showed

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/playlist-next-screen/playlist-next-screen.js [hunk 1]
```diff
@@ -38,13 +38,28 @@ const COMPONENT_NAME = 'PlaylistNextScreen';
...
+
+  /**
+   * after component updated, if was renderd and now should then focus for keyboard a11y
+   * @param {Object} previousProps - previous props
+   * @returns {void}
+   * @memberof PlaylistCountdownPopup
+   */
+  componentDidUpdate(previousProps: Object): void {
+    if (!this._shouldRender(previousProps) && this._shouldRender(this.props)) {
+      if (this.focusElement) {
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/components/playlist-next-screen/playlist-next-screen.js]: fixed

File: src/components/playlist-next-screen/playlist-next-screen.js [hunk 2]
```diff
@@ -38,13 +38,28 @@ const COMPONENT_NAME = 'PlaylistNextScreen';
...
+  focusElement: HTMLElement;
+
+  /**
+   * after component updated, if was renderd and now should then focus for keyboard a11y
+   * @param {Object} previousProps - previous props
+   * @returns {void}
+   * @memberof PlaylistCountdownPopup
+   */
+  componentDidUpdate(previousProps: Object): void {
+    if (!this._shouldRender(previousProps) && this._shouldRender(this.props)) {
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/components/playlist-next-screen/playlist-next-screen.js]: Cannot use didMount as the didMount is fired when the ui is loaded even if the component is really rendered (shouldRender is false)

File: src/components/playlist-next-screen/playlist-next-screen.js [hunk 3]
```diff
@@ -38,13 +38,28 @@ const COMPONENT_NAME = 'PlaylistNextScreen';
  * @extends {Component}
  */
 class PlaylistNextScreen extends Component {
+  focusElement: HTMLElement;
+
+  /**
+   * after component updated, if was renderd and now should then focus for keyboard a11y
+   * @param {Object} previousProps - previous props
+   * @returns {void}
+   * @memberof PlaylistCountdownPopup
```
[the remark this answers is not in the document]
Comment (RoyBregman) [reply src/components/playlist-next-screen/playlist-next-screen.js]: fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,21 @@
 
 All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
 
+<a name="0.49.0"></a>
```
Comment (OrenMe) [inline CHANGELOG.md]: this shouldn't be here - did you run build script locally?

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@playkit-js/playkit-js-ui",
-  "version": "0.48.1",
+  "version": "0.49.0",
```
Comment (OrenMe) [inline package.json]: this shouldn't be here - did you run build script locally?

File: src/components/playlist-countdown/playlist-countdown.js [hunk 6]
```diff
@@ -161,6 +167,7 @@ class PlaylistCountdown extends Component {
     const className = [style.playlistCountdown];
     const isHidden = !this.state.timeToShow || countdown.duration >= props.duration;
     const isCanceled = this.props.countdownCanceled;
+    this.setState({isHiddenOrCanceled: isHidden || isCanceled});
```
Comment (yairans) [inline src/components/playlist-countdown/playlist-countdown.js]: I would change the state to `show` (invert the condition for sure)

File: src/components/playlist-next-screen/playlist-next-screen.js [hunk 7]
```diff
@@ -38,13 +38,26 @@ const COMPONENT_NAME = 'PlaylistNextScreen';
...
+    }
+  }
   /**
    * should render component
    * @param {*} props - component props
    * @returns {boolean} - component element
    */
   _shouldRender(props: any): boolean {
-    return props.playlist && props.playlist.next && props.playlist.next.sources;
+    return !!props.playlist && !!props.playlist.next && !!props.playlist.next.sources && props.isPlaybackEnded;
```
Comment (yairans) [inline src/components/playlist-next-screen/playlist-next-screen.js]: Do it once:
```suggestion
    return !!(props.playlist && props.playlist.next && props.playlist.next.sources && props.isPlaybackEnded);
```

Comment (RoyBregman) [reply src/components/playlist-next-screen/playlist-next-screen.js]: fixed

File: src/components/playlist-countdown/playlist-countdown.js (hunk 6 above)
Comment (RoyBregman) [reply src/components/playlist-countdown/playlist-countdown.js]: fixed

**Pull request merged**